dbt_subprojects/dex/macros/models/_decoding/uniswap_v3/uniswap_v3_trades.sql

```diff
@@ -0,0 +1,85 @@
+
+{% macro uniswap_v3_forks_trades(
+    blockchain = null
+    , dex_type = 'uni-v3'
+    , project = null
+    , version = null
+    , Pair_evt_Swap = null
+    , Factory_evt_PoolCreated = null
+    , pair_column_name = 'pool'
+    , taker_column_name = 'recipient'
+    , maker_column_name = null
+    )
+%}
+
+WITH evt_swap AS (
+    SELECT
+        block_number
+        , block_time
+        , {{ taker_column_name }}
+        {% if maker_column_name %}
+        , {{ maker_column_name }}
+        {% endif %}
+        , amount0
+        , amount1
+        , contract_address
+        , tx_hash
+        , index
+    FROM {{ Pair_evt_Swap }}
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
+)
+
+, dexs AS
+(
+    SELECT
+        t.block_number
+        ,t.block_time
+        , t.{{ taker_column_name }} AS taker
+        , {% if maker_column_name %}
+                t.{{ maker_column_name }}
+            {% else %}
+                cast(null as varbinary)
+            {% endif %} as maker
+        , CASE WHEN amount0 < INT256 '0' THEN abs(amount0) ELSE abs(amount1) END AS token_bought_amount_raw -- when amount0 is negative it means trader_a is buying token0 from the pool
+        , CASE WHEN amount0 < INT256 '0' THEN abs(amount1) ELSE abs(amount0) END AS token_sold_amount_raw
+        , CASE WHEN amount0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
+        , CASE WHEN amount0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
+        , t.contract_address as project_contract_address
+        , t.tx_hash
+        , t.index as evt_index
+        , f.contract_address as factory_address
+    FROM
+        evt_swap t
+    INNER JOIN
+        {{ Factory_evt_PoolCreated }} f
+        ON f.{{ pair_column_name }} = t.contract_address
+    --some scammers emitted events with established pair addresses, joining in the the creation trace to resolve to correct factory
+    INNER JOIN {{ source(blockchain, 'creation_traces') }} ct 
```

Comment:
can / should this be filtered incrementally in the join?

Reply:
Edge cases like new trade events from old pools.